tests/Metrics/MetricsTest.cpp

```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
   T zdo = .1, zero = 0., one = 1.;
   Gauge g(zero, "gauge_1", "Gauge 1");
 
+  using namespace std;
+  using namespace std::chrono;
+
+  random_device rnd_device;
+  mt19937 mersenne_engine {rnd_device()};  // Generates random integers
+  uniform_real_distribution<T> dist {T(1), T(100)};
+  vector<T> vr(1000);
+  auto gen = [&dist, &mersenne_engine](){
+               return dist(mersenne_engine);
+             };
+  generate(vr.begin(), vr.end(), gen);
+
+  auto start = high_resolution_clock::now();
+  size_t const p = 10;
+  size_t const part = vr.size()/p;
+  std::vector<std::future<void>> f;
```

Comment:
I guess you want the lambdas started as fast as possible, so:
```suggestion
  std::vector<std::future<void>> f;
  f.reserve(p);
```